src/py/mat3ra/made/tools/analyze.py

```diff
@@ -89,3 +90,63 @@ def get_closest_site_id_from_position(material: Material, position: List[float])
     position = np.array(position)  # type: ignore
     distances = np.linalg.norm(coordinates - position, axis=1)
     return int(np.argmin(distances))
+
+
+def select_atoms_within_layers(material: Material, atom_index: int, layer_thickness: float):
```

Comment:
This should be called `get_atom_indices_within_layer_containing_atom_by_index`

We should also add a helper ``get_atom_indices_within_layer_containing_atom_by_position` in case we just want to pass position and then find the closest atom to it

Reply:
get_atom_indices_within_layer_by_atom_index
get_atom_indices_within_layer_by_atom_position
   finds index first, uses get_atom_indices_within_layer_containing_atom_by_index

`get_atom_indices_within_layer` to do both

`get_atom_indices_within_layer(id)`
`get_atom_indices_within_layer(position=[0, 0, 0])`